feature: add TRIM function and documentation (#1255)
nebulastream/nebulastream: docs/guide/how_to_add_a_function.md

> # How to add a `Function` in NebulaStream

Reviewer: I would like to have this document in a separate commit. If we want to revert some changes for the trim function, we would also revert this document. This would introduce additional work then

Functions in NebulaStream are used to transform data within query plans. They can operate on various data types and are used in SQL expressions like `SELECT TRIM(name) FROM stream`.

Functions in NebulaStream follow a two level architecture:
1. **Logical Functions**: represent functions in the logical query plan, handling type inference and validation.
2. **Physical Functions**: execute the actual computation at runtime using the Nautilus.

> Below guide will walk you through implementing a function using the `TRIM` function as an example. TRIM removes leading and trailing whitespace from variable-sized strings (VARSIZED).

Reviewer: ```suggestion
The guide below walks you through implementing a function using `TRIM` as an example. TRIM removes leading and trailing whitespace from variable-sized strings (VARSIZED).
```

Maybe you can add here a link to the c++ trim function

## 1. Overview

Functions are implemented in two locations:
- **Logical functions**: `nes-logical-operators/src/Functions/` and `nes-logical-operators/include/Functions/`
- **Physical functions**: `nes-physical-operators/src/Functions/` and `nes-physical-operators/include/Functions/`

For our TRIM example, we create:
```
nes-logical-operators/
|── include/Functions/
    ── TrimLogicalFunction.hpp
|── src/Functions/
    ── TrimLogicalFunction.cpp

nes-physical-operators/
|── include/Functions/
    ── TrimPhysicalFunction.hpp
|── src/Functions/
    ── TrimPhysicalFunction.cpp
```

As a last step before build, you'll need to:
- register the function in CMakeLists.txt files

> - create system tests in `nes-systests/function/`

Reviewer: ```suggestion
As a last step before building, you'll need to:
- register the function in CMakeLists.txt files
- create system tests in `nes-systests/function/`
```

## 2. Logical Function 

A logical function class must inherit from `LogicalFunctionConcept` and implement all required methods. Reference existing functions like `ConcatLogicalFunction` or `TrimLogicalFunction` for complete examples.

### 2.1 Key Components

**Header file** (`TrimLogicalFunction.hpp`):
- inherit from `LogicalFunctionConcept`

> - define `static constexpr std::string_view NAME = "Trim"` (this should match SQL function name)

Reviewer: ```suggestion
- define `static constexpr std::string_view NAME = "Trim"` (this should match the SQL function name)
```
